rdrf/rdrf/views/form_view.py

```diff
@@ -1565,10 +1565,16 @@ class FileUploadView(View):
 
     @login_required_method
     def get(self, request, registry_code, file_id):
-        data, filename = filestorage.get_file(file_id)
-        if data is not None:
-            response = FileResponse(data, content_type='application/octet-stream')
-            response['Content-disposition'] = 'filename="%s"' % filename
+        file_info = filestorage.get_file(file_id)
+        if file_info.patient:
+            security_check_user_patient(request.user, file_info.patient)
+        elif file_info.uploaded_by:
+            if user_is_patient_type(request.user) and request.user != file_info.uploaded_by:
+                raise PermissionDenied
+
```

Comment:
I'm not sure about the checks above.

Isn't the file always associated with a `Patient`? In what circumstances will we have a file that doesn't have a patient?

Depending on the answer for the question above is the user check needed? I don't think I like it, it could lead to subtle bugs. Ie. user uploads a file for a patient as a curator then it gets removed from the curator groups, we will still allow them to download the file.

And finally, we should default to `PermissionDenied` if there isn't enough information to decide whether we should allow the download. This is a security fix, so it is better to be stricter but safe. Ie. not enough file information -> `PermissionDenied`.

Reply:
I removed the user check and updated the code to raise `PermissionDenied` if patient is not set in CDEFile.